### PR TITLE
customize flumecodec.json in migrate.js to support bendybutt

### PR DESCRIPTION
**Pull request by staltz**

### CODE REVIEW

File: test/migration.js [hunk 1]
```diff
@@ -20,14 +20,91 @@ const dir = '/tmp/ssb-db2-migrate'
 rimraf.sync(dir)
 mkdirp.sync(dir)
 
-const TOTAL = 10
+const M = 10
+const A = 5
```
Comment (arj03) [inline test/migration.js:24]: Back to single letter variables. Just kidding ;-)

**Comment by arj03** [comment]
It seems to have failed the test?

**Comment by staltz** [comment]
Yeah, the failure I see in CI is totally new, not matching what I get locally (even if I install node_modules from scratch)

File: migrate.js [hunk 2]
```diff
@@ -31,6 +30,30 @@ function makeFileExistsObv(filename) {
...
+// bendy butt messages which may contain Buffers
+const jsonCodec = {
+  encode: JSON.stringify,
+  decode(str) {
+    const parsed = JSON.parse(str)
+    const content = parsed.value.content
+    if (
+      typeof content === 'object' &&
+      content.type === 'metafeed/add/derived'
+    ) {
```
Comment (staltz) [inline migrate.js:43]: Hopefully this `if` condition will preserve the performance of the jsonCodec since `content.type` is rarely add/derived

**Comment by arj03** [comment]
Migrate (+db1) is a lot higher?

**Comment by staltz** [comment]
I added a small optimization https://github.com/ssb-ngi-pointer/ssb-db2/pull/270/commits/ad47efefe7c4e7bdb5ca581dae3d1dc84c36ef10 but it didn't seem to have meaningful impact.

I don't know how to improve this, and it seems like we need the fix merged.

@arj03 ?

**Comment by staltz** [comment]
Maybe we can add a `config.db2._fixturesHacks` that activates this codec only for ssb-fixtures? It seems contrived/rare/impossible that a flumelog would have a metafeed message in it, in real world use cases.

**Comment by staltz** [comment]
Okay, now I confess I don't understand this anymore. :sweat_smile:

**Comment by staltz** [comment]
Benchmarks ran locally:

## Before

```
  generate fixture with flumelog-offset

    ✔ seed = sloop
    ✔ messages = 100000
    ✔ authors = 2000
    ✔ log.offset was created

  setup


  add a bunch of messages

    ✔ delete db2 folder to start clean
    ✔ duration: 510ms

  migrate (+db1)

    ✔ delete db2 folder to start clean
    ✔ ssb-db has finished indexing
    ✔ duration: 24746ms
    fallback to close

  migrate (alone)

    ✔ delete db2 folder to start clean
    ✔ duration: 4103ms

  migrate (+db1 +db2)

    ✔ delete db2 folder to start clean
    ✔ duration: 5739ms
    fallback to close

  migrate (+db2)

    ✔ delete db2 folder to start clean
    ✔ duration: 5567ms

  migrate continuation (+db2)

    ✔ delete db2 folder to start clean
    ✔ migrated 90%, will reset sbot
    ✔ duration: 842ms

  Memory usage without indexes

    ✔ memory usage without indexes: 714.58 MB = 28.01 MB + etc

  initial indexing

    ✔ null
    ✔ duration: 716ms

  initial indexing maxcpu 86

    ✔ delete indexes folder to start clean
[Error: ENOENT: no such file or directory, rename '/tmp/ssb-db2-benchmark/db2/indexes/value_content_type_post.index~' -> '/tmp/ssb-db2-benchmark/db2/indexes/value_content_type_post.index'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'rename',
  path: '/tmp/ssb-db2-benchmark/db2/indexes/value_content_type_post.index~',
  dest: '/tmp/ssb-db2-benchmark/db2/indexes/value_content_type_post.index'
}
    ✔ null
    ✔ duration: 4918ms

  initial indexing compat

    ✔ duration: 817ms

  Two indexes updating concurrently

    ✔ duration: 630ms

  setup


  key one initial

    ✔ duration: 47ms

  key two

    ✔ duration: 1ms

  key one again

    ✔ duration: 1ms

  reboot

loadStateFeeds failed: Error: Closed

  reboot and key one again

    ✔ duration: 49ms

  latest root posts

    ✔ duration: 474ms

  latest posts

    ✔ duration: 29ms

  votes one initial

    ✔ duration: 449ms

  votes again

    ✔ duration: 1ms

  hasRoot

    ✔ duration: 286ms

  hasRoot again

    ✔ duration: 1ms

  author one posts

    ✔ duration: 223ms

  author two posts

    ✔ duration: 16ms

  dedicated author one posts

    ✔ duration: 259ms

  dedicated author one posts again

    ✔ duration: 2ms

  maximum RAM used

    ✔ maximum memory usage: 802.59 MB = 49.58 MB + etc

  Indexes folder size

    ✔ indexes folder size: 12.80mb

  teardown



  total:     42
  passing:   42
  duration:  3m 33.4s

```

## After

```

  generate fixture with flumelog-offset

    ✔ seed = sloop
    ✔ messages = 100000
    ✔ authors = 2000
    ✔ log.offset was created

  setup


  add a bunch of messages

    ✔ delete db2 folder to start clean
    ✔ duration: 502ms

  migrate (+db1)

    ✔ delete db2 folder to start clean
    ✔ ssb-db has finished indexing
    ✔ duration: 10570ms
    fallback to close

  migrate (alone)

    ✔ delete db2 folder to start clean
    ✔ duration: 3927ms

  migrate (+db1 +db2)

    ✔ delete db2 folder to start clean
    ✔ duration: 7456ms
    fallback to close

  migrate (+db2)

    ✔ delete db2 folder to start clean
    ✔ duration: 5685ms

  migrate continuation (+db2)

    ✔ delete db2 folder to start clean
    ✔ migrated 90%, will reset sbot
    ✔ duration: 806ms

  Memory usage without indexes

    ✔ memory usage without indexes: 724.62 MB = 30.35 MB + etc

  initial indexing

    ✔ null
    ✔ duration: 735ms

  initial indexing maxcpu 86

    ✔ delete indexes folder to start clean
[Error: ENOENT: no such file or directory, rename '/tmp/ssb-db2-benchmark/db2/indexes/value_content_type_post.index~' -> '/tmp/ssb-db2-benchmark/db2/indexes/value_content_type_post.index'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'rename',
  path: '/tmp/ssb-db2-benchmark/db2/indexes/value_content_type_post.index~',
  dest: '/tmp/ssb-db2-benchmark/db2/indexes/value_content_type_post.index'
}
    ✔ null
    ✔ duration: 4862ms

  initial indexing compat

    ✔ duration: 793ms

  Two indexes updating concurrently

    ✔ duration: 735ms

  setup


  key one initial

    ✔ duration: 53ms

  key two

    ✔ duration: 1ms

  key one again

    ✔ duration: 2ms

  reboot

loadStateFeeds failed: Error: Closed

  reboot and key one again

    ✔ duration: 47ms

  latest root posts

    ✔ duration: 443ms

  latest posts

    ✔ duration: 46ms

  votes one initial

    ✔ duration: 495ms

  votes again

    ✔ duration: 1ms

  hasRoot

    ✔ duration: 307ms

  hasRoot again

    ✔ duration: 2ms

  author one posts

    ✔ duration: 250ms

  author two posts

    ✔ duration: 18ms

  dedicated author one posts

    ✔ duration: 251ms

  dedicated author one posts again

    ✔ duration: 2ms

  maximum RAM used

    ✔ maximum memory usage: 810.46 MB = 46.96 MB + etc

  Indexes folder size

    ✔ indexes folder size: 12.80mb

  teardown



  total:     42
  passing:   42
  duration:  2m 58.4s

```

**Pull request merged**